atomix/cluster/src/main/java/io/atomix/raft/storage/log/RaftLogUncommittedReader.java

```diff
@@ -0,0 +1,84 @@
+/*
+ * Copyright 2017-present Open Networking Foundation
+ * Copyright © 2020 camunda services GmbH (info@camunda.com)
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.atomix.raft.storage.log;
+
+import io.atomix.raft.storage.log.entry.RaftLogEntry;
+import io.atomix.raft.storage.serializer.RaftEntrySBESerializer;
+import io.atomix.raft.storage.serializer.RaftEntrySerializer;
+import io.camunda.zeebe.journal.JournalReader;
+import io.camunda.zeebe.journal.JournalRecord;
+import java.util.NoSuchElementException;
+
+/**
+ * Raft log reader that reads both committed and uncommitted entries. This reader is supposed to be
+ * only used internally in raft. The reader is not thread safe with a concurrent writer.
+ */
+public class RaftLogUncommittedReader implements RaftLogReader {
```

Comment:
:thought_balloon: I feel the name might be misleading. I might read this and think it reads only the uncommitted entries.

Maybe we can find a better name. Something like `RaftEagerReader` or `RaftLenientReader`, but as you can see, no obvious name comes to my mind, either

Reply:
For me uncommittedReader sounds good. `Read Uncommitted` is a commonly used isolation level in databases which indicates it also allows to read uncommitted data, besides everything that is possible in stricter isolation levels. 

Alternate name I can think of is `UnrestrictedReader` and `CommittedReader`. What do you think?

Reply:
I like the analogy with databases better.